javatests/arcs/android/systemhealth/testapp/StorageCore.kt

```diff
@@ -278,73 +280,93 @@ class StorageCore(val context: Context, val lifecycle: Lifecycle) {
         settings: Settings
     ) = when (settings.handleType) {
         HandleType.SINGLETON -> {
-            taskHandle.handle = (
-                taskHandle.handleManager.createHandle(
-                    HandleSpec(
-                        "singletonHandle$taskId",
-                        HandleMode.ReadWrite,
-                        HandleContainerType.Singleton,
-                        TestEntity.Companion
-                    ),
-                    when (settings.storageMode) {
-                        TestEntity.StorageMode.PERSISTENT -> TestEntity.singletonPersistentStorageKey
-                        else -> TestEntity.singletonInMemoryStorageKey
+            val handle = taskHandle.handleManager.createHandle(
+                HandleSpec(
+                    "singletonHandle$taskId",
+                    HandleMode.ReadWrite,
+                    HandleContainerType.Singleton,
+                    TestEntity.Companion
+                ),
+                when (settings.storageMode) {
+                    TestEntity.StorageMode.PERSISTENT -> TestEntity.singletonPersistentStorageKey
+                    else -> TestEntity.singletonInMemoryStorageKey
+                }
+            ) as ReadWriteSingletonHandle<TestEntity>
+
+            taskHandle.handle = withContext(handle.dispatcher) {
+                val elapsedTime = measureTimeMillis { handle.awaitReady() }
```

Comment:
this also doesn't need to be called within the dispatcher... awaitReady registers an onReady listener.